src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java

```diff
@@ -143,8 +143,21 @@ public void load() throws IOException {
      */
     public void persist() throws IOException {
         final Path path = Paths.get(fileName);
-        final Path directory = path.getParent();
+        Path directory = path.getParent();
+
         if (directory != null) {
+            if (Files.isSymbolicLink(directory)) {
+                final Path actualDir = Files.readSymbolicLink(directory);
+
+                if (Files.isDirectory(actualDir)) {
+                    directory = actualDir;
+                }
+                else {
+                    throw new IOException(
+                            "Resolved symbolic link " + directory
+                                    + " does not exist or is not a directory.");
```

Comment:
>does not exist

How it can be possible if we do `(Files.isSymbolicLink(directory))`?
If not possible, let's remove this phrase.

Reply:
if `Files.isSymbolicLink(directory)` returns true, it means that there is indeed a symbolic link at the directory path. But, what if symbolic link is broken (pointing to a non-existing target) or the target is not a directory.
Therefore, i have a check i.e. Files.isDirectory(actualDir) to see if the link points to a directory or not. If not, throws an IOException because the symbolic link does not point to a valid directory.

That was my intent. Please let me know if i understood the review suggestion right.

Reply:
> if `Files.isSymbolicLink(directory)` returns true, it means that there is indeed a symbolic link at the directory path. But, what if symbolic link is broken (pointing to a non-existing target) or the target is not a directory. Therefore, i have a check i.e. `Files.isDirectory(actualDir`) to see if the link points to a directory or not. If not, throws an `IOException` because the symbolic link does not point to a valid directory.
> 
> That was my intent. Please let me know if i understood the review suggestion right.

Reply:
>But, what if symbolic link is broken (pointing to a non-existing target)

We are not checking it. We do check only that it is not a directory. So message should match to what we do

Reply:
Okay, understood. We are not checking if it exists. And, the message should look something like this: ` "Resolved symbolic link " + directory + "is not a directory."`

Reply:
Yes